### DISCUSSION TITLE
Mercure scaling + clustering in production.

### DISCUSSION THREAD

**Discussion by danmatthews** [php/frankenphp#421] [Q&A] [answered]
Like a few folk, i'm intrigued by the built in Mercure hub and the possibility of using it with Laravel without having to run Soketi, Pusher, or our own Laravel Echo server.

My main questions are

1) If i'm in a load-balanced setup, can the instances communicate with each other via the load balancer, or would internal connections and clustering have to be configured?

2) Similar to the above, if a background process is running on a separate queue-worker server (a queue job in Laravel), and I broadcast a message intended for a user listening on one of the web servers, how would that reach the user? would i have to have a clustering situation set up as above?

3) Would triggering auto-scaling based on load still be a proficient strategy for bringing up new instances when the instance FrankenPHP is running on is struggling because of _either_ the PHP or Mercure processes taking a hammering?

Not looking for specific Devops advice, but some opinions / best practices and some links to the relevant documentation would be most welcome.

Thanks for all the hard work on this gents, i'm glad it's getting some traction within the community, it deserves it.

**Comment by dunglas** [comment, marked as answer]
Hi,

1. Mercure supports load-balancing, but only in the paid version (SaaS or On-Prem, it's how we fund the project and various open-source work) or by writing a custom transport: https://mercure.rocks/docs/hub/cluster
2. In case you have a version supporting clustering, you can hit any server from your worker, and the message will be dispatched to all instances
3. Yes for FrankenPHP, for Mercure, yes if you have a version supporting clustering
